site-src/guides/implementers.md

```diff
@@ -159,16 +159,118 @@ the `metadata.generation` field of the object at the time the status is generate
+ensure this behavior is consistent across implementations that claim support for
+the feature.
+
+### Indirect Configuration
+There are a variety of instances where TLS Certificates may not be managed
+directly by the owner of a Gateway. Although this is not meant to be an
+exhaustive list, it documents some of the approaches we expect to see used to
+manage TLS Certificates with Gateway API:
+
+#### 1. Certificates from other places
+Some providers offer the ability to configure and host TLS Certificates outside
+of Kubernetes altogether. Implementations that can connect to those external
+providers may expose that capability via a TLS option on the Gateway Listener,
+for example:
+
+```
+  listeners:
+  - name: https
+    protocol: HTTPS
+    port: 443
+    tls:
+      mode: Terminate
+      options:
+        vendor.example.com/certificate-name: store-example-com
+```
+
+In this example, the `store-example-com` name would refer to the name of a
+certificate stored by the external `vendor.example.com` TLS Certificate
+provider.
+
+#### 2. Automatically generated TLS certs that are populated later
+Many users would prefer that TLS certs were automatically generated on their
+behalf. One potential implementation of that would involve a controller that
+watches Gateways and HTTPRoutes, generates TLS certs, and attaches them to the
+Gateway. Depending on the implementation details, Gateway owners may need to
+configure something at the Gateway or Listener level to explicitly opt-in to
+this feature. For example, let's say someone created `acme-cert-generator` to
+generate certs following this pattern. That generator may choose to only
+generate and populate certs on Gateway Listeners with `acme.io/cert-generator`
+set in `tls.options` or a similar annotation set for the entire Gateway.
+
+Note that this is actually fairly similar to [how Cert Manager works
+today](https://cert-manager.io/docs/usage/gateway/), but that requires Gateway
+owners to reference a Kubernetes Secret that it will then populate. This
+specific approach was required because TLS CertificateRefs were required to be
+specified until Gateway API v1.1.
+
+With the relaxing of Gateway API validation in v1.1, TLS Certificates can be
+left unspecified on creation, allowing for less configuration when working with
+generated TLS certificates.
+
+#### 3. Certs that are specified by other personas
+In some organizations, Application Developers are responsible for managing TLS
+Certificates (see [Roles and Personas](/concepts/roles-and-personas) for more on
+this and other roles).
+
+To enable this use case, a new controller and CRD would be created. This
+CRD would link hostnames to user-provided certs, and then the controller would
+populate the certs specified by that CRD on Gateway listeners that matched those
+hostnames. This would also likely benefit from a Listener or Gateway-level
+opt-in for the behavior.
```

Comment:
Would an option be a bad example of a way to offer opt-in behavior here? It would definitely be a bit of an abuse of the `map[string]string` structure, but would have the benefit of not requiring API changes.


For example:

`example.com/tls-allowed-namespace-selector: foo=bar`

Reply:
I'm generally against any form of structured content within these kind of annotation-like fields since it's so easy to mess up. Maybe I should add some guidelines for what doesn't belong here? Was that what you were suggesting here?